backend/fms_core/migrations/0019_v3_1_2.py

```diff
@@ -0,0 +1,65 @@
+from django.conf import settings
+from django.db import migrations
+from django.contrib.auth.models import User
+import reversion
+import datetime
+
+ADMIN_USERNAME='biobankadmin'
+
+
+def fix_process_data(apps, schema_editor):
+    Version = apps.get_model("reversion", "Version")
+    Revision = apps.get_model("reversion", "Revision")
+    ProcessSample = apps.get_model("fms_core", "ProcessSample")
+    Process = apps.get_model("fms_core", "Process")
+
+    affected_revisions_ids = Revision.objects.filter(date_created__range=["2021-04-29", "2021-05-01"]).values_list('id', flat=True)
+
+    with reversion.create_revision(manage_manually=True):
+        reversion.set_comment("Curation of multiple processes into one process per template")
+        reversion.set_user(User.objects.get(username=ADMIN_USERNAME))
+
+        for revision_id in affected_revisions_ids:
+            # Getting all ProcessSample created in this revision
+            affected_process_samples_ids = Version.objects.filter(serialized_data__contains='"model": "fms_core.processsample"',
+                                                   revision_id=revision_id).values_list('object_id', flat=True)
+
+            if affected_process_samples_ids.count() > 1:
+                process_samples = ProcessSample.objects.filter(id__in=list(affected_process_samples_ids))
+
+                # Getting all Processes related to those ProcessSample
+                process_ids = list(process_samples.values_list('process_id', flat=True))
+
+                if len(process_ids) > 1:
+                    # First Process; the only one that will be kept
+                    first_process_id = min(process_ids)
+
+                    # Change process_id for ProcessSample now pointing to the wrong processes
+                    for ps in process_samples:
+                        ps.process_id = first_process_id
+                        ps.save()
+                        reversion.add_to_revision(ps)
+
+                    # Delete Processes unused (should not have been created in first place)
+                    Process.objects.filter(pk__in=process_ids).exclude(pk=first_process_id).delete()
```

Comment:
This is likely to call a bulk_delete, not sure this will call our delete (not saving a version for processes). I am not sure we want to save a deleted version of those processes, but we might want to since we had an initial version. It would be better for traceability too. Iterating over the processes instead to access the model delete instead of the queryset delete might work better.

Reply:
Right.. I was wondering whether we should version those processes to be deleted, it might be confusing when we have new valid processes with the same ids as those that were deleted.. But I guess a way to make a compromise would be to manually change the comment of those processes before deleting them manually too..